### PR TITLE
[Different] feature/SCKT-1_different_init_project_supaapps_api_kit_client

### PR DESCRIPTION
**Pull request by MostafaFawzy7**

### CODE REVIEW

File: src/ApiKitClient.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import axios, { AxiosError, AxiosRequestConfig } from 'axios';
+import { HttpMethod, ApiClientResponse, PaginatedResponse } from './types';
+
+export function ApiKitClient<T>(
+  baseUrl: string,
+  authToken?: string,
+  handleUnauthorizedAccess?: () => void
+) {
```
Comment (Abdallah-Abdelkhalek) [inline src/ApiKitClient.ts:8]: authToken optional ?

File: src/ApiKitClient.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      const response = await axiosInstance.request(config);
+      return { status: response.status, data: response.data };
+    } catch (error) {
+      const axiosError = error as AxiosError | any;
+      return {
+        status: axiosError.response?.status || 500,
+        errors: axiosError.response?.data.errors,
+      };
+    }
+  };
```
Comment (Abdallah-Abdelkhalek) [inline src/ApiKitClient.ts:46]: where is the logic for `handleUnauthorizedAccess`

Comment (MostafaFawzy7) [reply src/ApiKitClient.ts:46]: This is the point! (To not be logic here)...
As, we agreed on the call with Nawwar, that the logic of unauthorization should be a callback to allow the consumer app to determine what is the logic should be, for ex:

```js
Router.push('/login') // in React or Next
```

File: src/ApiKitClient.ts (hunk 1 above)
Comment (MostafaFawzy7) [reply src/ApiKitClient.ts:8]: I think I missed this!
We have to agree on if the unauthorization callback will be optional as well or not!

File: src/ApiKitClient.ts (hunk 2 above)
Comment (Abdallah-Abdelkhalek) [reply src/ApiKitClient.ts:46]: aha so the logic is not handled in this app

Comment (MostafaFawzy7) [reply src/ApiKitClient.ts:46]: Exactly!

**Comment by SirNarsh** [comment]
we prefer the other with singleton

**Comment by MostafaFawzy7** [comment]
Closing this PR and go with the other approach!

**Pull request closed without merging**